httprequest_lego_provider/api/dns.py

```diff
@@ -68,58 +44,41 @@ def _line_matches_subdomain(line: str, subdomain: str) -> bool:
     return not line.strip().startswith(";") and bool(line.split()) and line.split()[0] == subdomain
 
 
-def _remove_subdomain_entries_from_file_content(
-    content: Iterable[str], subdomain: str
-) -> List[str]:
-    """Remove from the file the entries matching a subdomain.
-
-    Args:
-        content: the file content.
-        subdomain: the subdomain for which to filter out the entries.
-
-    Returns:
-        the content excluding the entries for the  subdomain.
-    """
-    new_content = []
-    for line in content:
-        if not _line_matches_subdomain(line, subdomain):
-            new_content.append(line)
-        else:
-            logging.error("Subdomain %s already present as a DNS record.", subdomain)
-    return new_content
-
-
-def write_dns_record(fqdn: str, value: str) -> None:
+def write_dns_record(fqdn: str, rdata: str) -> None:
     """Write a DNS record.
 
     Args:
         fqdn: the FQDN for which to add a record.
-        value: ACME challenge for DNS record to add.
+        rdata: ACME challenge for DNS record to add.
 
     Raises:
         DnsSourceUpdateError: if an error while updating the repository occurs.
     """
-    user, base_url, branch = parse_repository_url(GIT_REPO_URL)
-    with TemporaryDirectory() as tmp_dir:
-        try:
-            repo = Repo.clone_from(base_url, tmp_dir, branch=branch)
-            config_writer = repo.config_writer()
-            config_writer.set_value("user", "name", user)
-            config_writer.release()
-            domain, subdomain = _get_domain_and_subdomain_from_fqdn(fqdn)
-            filename = FILENAME_TEMPLATE.format(domain=domain)
-            dns_record_file = Path(f"{repo.working_tree_dir}/{filename}")
-            content = dns_record_file.read_text("utf-8")
-            new_content = _remove_subdomain_entries_from_file_content(
-                io.StringIO(content), subdomain
-            )
-            new_content.append(RECORD_CONTENT.format(record=subdomain, value=value))
-            dns_record_file.write_text("".join(new_content), encoding="utf-8")
-            repo.index.add([filename])
-            repo.git.commit("-m", f"Add {fqdn} record")
-            repo.remote(name="origin").push()
-        except (GitCommandError, ValueError) as ex:
-            raise DnsSourceUpdateError from ex
+    try:
+        # Notify the charm that we need to request a DNS record
+        output = subprocess.check_output(
+            ["/usr/bin/pebble", "notify", "dns.local/write", f"fqdn='{fqdn}'", f"rdata='{rdata}'"],
+            stderr=subprocess.STDOUT,  # nosec B603, B607
+            timeout=10,
+        )
+
+        if b"Error" in output or b"error" in output:
+            raise DnsSourceUpdateError(f"Error executing Pebble command: {output.decode('utf-8')}")
```

Comment:
The errors of the API will be sent back to lego and printed in the lego logs. So we need to have a comprehensive message for the operator that will read those logs